corehq/ex-submodules/casexml/apps/phone/models.py

```diff
@@ -633,9 +634,9 @@ def _sync_log_was_old():
                         this_case_index not in candidates_to_remove):
                     self.prune_case(this_case_index)
         else:
-            # we have some possible candidates for removal. we should check each of them.
-            candidates_to_remove.remove(case_id)  # except ourself
-            for candidate in candidates_to_remove:
+            # we have some possible candidates for removal. we should check each of them except ourself
+            candidates_to_also_check = candidates_to_remove - set(case_id)
```

Comment:
the issue was that `candidates_to_remove` is used in a closure inside `_remove_case` above. it would be nice to get rid of the closures altogether since they're causing issues, but i figured I would hold off on any major changes until extension cases is merged in